packages/compiler/src/core/index.ts

```diff
@@ -32,5 +32,7 @@ export {
   Queue,
   TwoLevelMap,
   createRekeyableMap,
+  deepClone,
+  deepEquals,
```

Comment:
@bterlson what do you think of exporting those directly from the compiler, I feel like we should try to not keep exposing generic utils right from the root.

Reply:
"@typspec/compiler/utils" sub export

Reply:
Sorry also typed that without adding extra context. We decided yersteday to make a new `./utils` export for the compiler where we can expose those things. I'll make a separate PR. so you can revert that and instead use the import from `@typespec/compiler/utils`

Reply:
PR for this change is here, https://github.com/microsoft/typespec/pull/2919, just got to let our partner teams about this change so targeting to merge that around thursday this week.

Reply:
@susliko  I merged that PR you should be good to update and I think we can merge your PR

Reply:
@timotheeguerin I've rebased on its, squashed my commits and fixed linting&formatting errors